3452.xml

```diff
@@ -0,0 +1,167 @@
+	<Object ObjectType="MODefinition">
+		<Name>Program Scheduler</Name>
+		<Description1>Summary of the mapping between the various functions, eventual inputs and the target actuators, spanning multiple days or years.</Description1>
+		<ObjectID>3452</ObjectID>
+		<ObjectURN>urn:oma:lwm2m:ext:3452</ObjectURN>
+		<LWM2MVersion>1.0</LWM2MVersion>
+		<ObjectVersion>1.0</ObjectVersion>
+		<MultipleInstances>Multiple</MultipleInstances>
+		<Mandatory>Mandatory</Mandatory>
+		<Resources>
+			<Item ID="0">
+				<Name>Schedule Name</Name>
+				<Type>String</Type>
+				<Operations>RW</Operations>
+				<Mandatory>Optional</Mandatory>
+				<MultipleInstances>Single</MultipleInstances>
+				<Description><![CDATA[Unique key identifying the Schedule]]></Description>
+				<Units/>
+			</Item>
+			<Item ID="1">
+				<Name>State</Name>
+				<Type>Boolean</Type>
+				<Operations>RW</Operations>
+				<Mandatory>Mandatory</Mandatory>
+				<MultipleInstances>Single</MultipleInstances>
+				<Description><![CDATA[If disabled, the instance shall be ignored]]></Description>
+				<Units/>
+			</Item>
+			<Item ID="2">
+				<Name>Priority</Name>
+				<Type>Integer</Type>
+				<Operations>RW</Operations>
+				<Mandatory>Mandatory</Mandatory>
+				<MultipleInstances>Single</MultipleInstances>
+				<Description><![CDATA[Integer representing the priority of the program within all program object instances.
+Lower number (with sign) correspond to higher priorities.
+Higher priority programs have pre-eminence over lower priority program schedule to be running in parallel in same day.
+For example: Program #11 is scheduled to run every day between Jan 1st and Dec 31st with a priority 7
+Program #22 is scheduled to run every Saturday between March 1st and June 30th with priority 3
+Program #22 should apply for all Saturdays between March 1st and June 30th
+Program #11 for all other Saturdays in the year
+If compound operation is supported, functions are executed in a order based on this priority field]]></Description>
+				<Units/>
+			</Item>
+			<Item ID="3">
+				<Name>Calendar Rules</Name>
+				<Type>Integer</Type>
+				<Operations>RW</Operations>
+				<Mandatory>Optional</Mandatory>
+				<MultipleInstances>Multiple</MultipleInstances>
+				<Description><![CDATA[Number of Calendar Rule object(s) instance containing information on which days in the week, months, year(s) is the program scheduled to run.]]></Description>
+				<Units/>
+			</Item>
+			<Item ID="4">
+				<Name>Program Function</Name>
+				<Type>Integer</Type>
+				<Operations>RW</Operations>
+				<Mandatory>Mandatory</Mandatory>
+				<MultipleInstances>Single</MultipleInstances>
+				<Description><![CDATA[Number of Program Function object instance containing information on how to control the output(s) based on time and sensor input(s)]]></Description>
+				<Units/>
+			</Item>
+			<Item ID="5">
+				<Name>Input Sources</Name>
+				<Type>String</Type>
```

Comment:
Would CoRE link make sense here as type?

Reply:
Thank you very much Ari and good point!
That could indeed make sense and could be considered very good practice.

Allow me to check internally to the Alliance if we can consider such change, in order to have a better formal description of the object, despite breaking compatibility with LWM2M 1.0

As an intermediate approach, could also be possible to keep String data type using Core Link syntax.

This also applies to some other resources as well, by the way (as you also commented below).